src/context/WalletProvider/TallyHo/components/Connect.tsx

```diff
@@ -68,8 +71,10 @@ export const TallyHoConnect = ({ history }: TallyHoSetupProps) => {
       try {
         const deviceId = await wallet.getDeviceID()
 
-        if (provider?.chainId !== '0x1') {
-          throw new Error('walletProvider.tallyHo.errors.network')
+        // Switch to Mainnet if wallet is on any other chain
+        const chainId = await wallet.ethGetChainId?.()
+        if (bnOrZero(chainId).toString() !== CHAIN_REFERENCE.EthereumMainnet) {
+          await wallet.ethSwitchChain?.(bn(CHAIN_REFERENCE.EthereumMainnet).toNumber())
```

Comment:
Note: This is currently a no-op. I managed to get to a chainId other than 1, and thus to this block, by building Tally from source and enabling all the multichain dev flags:
```
SUPPORT_POLYGON=true
SUPPORT_ARBITRUM=true
SUPPORT_OPTIMISM=true
```
However, `wallet_switchEthereumChain`, while broadcasted, isn't actually picked by Tally yet.
Until this gets released in Tally, this is a dead block, as the condition will always evaluate to false. IMO, it's better to still keep it here for consistency.

Reply:
Can we add a condensed version of this comment as a version-controlled comment?